### PR TITLE
Fix msGraph proxy options work

### PR DESCRIPTION
**Pull request by feedmypixel**

### CODE REVIEW

File: src/helpers/ms-graph.js [hunk 1]
```diff
@@ -16,13 +16,16 @@ const msGraphPlugin = {
...
-        host: proxyAgent?.url.hostname,
-        port: proxyAgent?.url.port
-      }
-    }
+    const proxy = proxyAgent()
+    const agent = proxy?.agent
+    const options = agent
+      ? {
+          proxyOptions: {
+            host: agent.proxy.href,
```
Comment (flurdy) [inline src/helpers/ms-graph.js:24]: or `host: proxy.proxy`

Comment (flurdy) [reply src/helpers/ms-graph.js:24]: Don't change it.

Comment (flurdy) [reply src/helpers/ms-graph.js:24]: So `proxy.url....`

File: src/helpers/ms-graph.js [hunk 2]
```diff
@@ -16,13 +16,16 @@ const msGraphPlugin = {
...
+            port: agent.connectOpts.port
```
Comment (flurdy) [inline src/helpers/ms-graph.js:25]: Or `proxy.url.port`...

File: src/helpers/ms-graph.js (hunk 1 above)
Comment (flurdy) [reply src/helpers/ms-graph.js:24]: :shrug:

Comment (feedmypixel) [reply src/helpers/ms-graph.js:24]: This is whats available to use in the `agent` 

```
HttpsProxyAgent {
  _events: [Object: null prototype] {
    free: [Function (anonymous)],
    newListener: [Function: maybeEnableKeylog]
  },
  _eventsCount: 2,
  _maxListeners: undefined,
  options: { path: undefined },
  requests: [Object: null prototype] {},
  sockets: [Object: null prototype] {},
  freeSockets: [Object: null prototype] {},
  keepAliveMsecs: 1000,
  keepAlive: false,
  maxSockets: Infinity,
  maxFreeSockets: 256,
  scheduling: 'lifo',
  maxTotalSockets: Infinity,
  totalSocketCount: 0,
  proxy: URL {
    href: 'http://***/',
    origin: 'http://***',
    protocol: 'http:',
    username: '',
    password: '',
    host: '***',
    hostname: '***',
    port: '',
    pathname: '/',
    search: '',
    searchParams: URLSearchParams {},
    hash: ''
  },
  proxyHeaders: {},
  connectOpts: {
    ALPNProtocols: [ 'http/1.1' ],
    host: '***',
    port: 80
  },
  [Symbol(shapeMode)]: false,
  [Symbol(kCapture)]: false,
  [Symbol(AgentBaseInternalState)]: {}
}```

File: src/helpers/ms-graph.js (hunk 2 above)
Comment (feedmypixel) [reply src/helpers/ms-graph.js:25]: so no port on `agent.proxy.port` - see above agent comment. This is only available on `agent.connectOpts.port`. I checked https://www.npmjs.com/package/https-proxy-agent and we are doing everything correct when instantiating this lib

Comment (feedmypixel) [reply src/helpers/ms-graph.js:26]: Now referencing the `proxy.url.href` 👍

**Review by flurdy — APPROVED** [review]

**Pull request merged**